install/kubernetes/cilium/templates/_helpers.tpl

```diff
@@ -5,6 +5,33 @@ Create chart name and version as used by the chart label.
 {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
 {{- end }}
 
+{{/*
+Render full image name from given values, e.g:
+```
+image:
+  repository: quay.io/cilium/cilium
+  tag: v1.10.1
+  useDigest: true
+  digest: abcdefgh
+```
+then `include "cilium.image" .Values.image`
+will return `quay.io/cilium/cilium:v1.10.1@abcdefgh`
+*/}}
+{{- define "cilium.image" -}}
+{{- $digest := (.useDigest | default false) | ternary (printf "@%s" .digest) "" -}}
+{{- printf "%s:%s%s" .repository .tag $digest -}}
+{{- end -}}
+
+{{- define "cilium.criticalPriorityClass" -}}
+{{- $root := index . 0 -}}
+{{- $priorityClass := index . 1 -}}
+{{- if and (eq $root.Release.Namespace "kube-system") (semverCompare ">=1.10-0" $root.Capabilities.KubeVersion.Version) -}}
+  {{- $priorityClass }}
+{{- else if semverCompare ">=1.17-0" $root.Capabilities.KubeVersion.Version -}}
+  {{- $priorityClass }}
+{{- end }}
+{{- end -}}
```

Comment:
So with this template, you removed almost all uses of the `enableCriticalPriorityClass` value:
https://github.com/cilium/cilium/blob/master/install/kubernetes/cilium/values.yaml#L1550-L1555

Which I think is fine, as again in newer versions of Helm we can now use `--kube-version`. However, there is one use of of it left (which also relies on `.Values.Capabilities` that you removed):

https://github.com/cilium/cilium/blob/8c49b836cb101b897be95c06410125fb59cd18b5/install/kubernetes/cilium/templates/etcd-operator/cilium-etcd-operator-deployment.yaml#L98

Do you think it makes sense to replace that use as well, and remove `enableCriticalPriorityClass` entirely? And then also document it as being removed in the upgrade notes?

(I realize that the last use is in `cilium-etcd-operator` which you are probably planning to remove anyway, but I prefer if we were consistent even for code that is about to be removed)

Reply:
Oh, I miss `enableCriticalPriorityClass` check. I just update `cilium.priorityClass` with this check and few changes in usage. And `cilium-etcd-operator` is updated as well.

For `enableCriticalPriorityClass` removal, I totally agree, but let ask other people before we make the change.

Reply:
Great change! Yes, agreed, let's keep `enableCriticalPriorityClass` in, now that it is used consistently. We can always remove it later.